### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by enriquebelarte**
Add e2e tests to prow. e2e-incluster-build.

### CODE REVIEW

File: ci/module-kmm-ci-build.template.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
         moduleName: kmm_ci_a
       kernelMappings:
         - literal: KVER_CHANGEME
-          containerImage: registry.minikube/kmm-kmod:local
+          containerImage: registry.ci.openshift.org/rh-ecosystem-edge/kmm-kmod:local
```
Comment (qbarrand) [inline ci/module-kmm-ci-build.template.yaml:13]: I think we rather want to push the image to the internal OpenShift Registry?
```suggestion
          containerImage: image-registry.openshift-image-registry.svc:5000/default/kmm-kmod:local
```

File: ci/module-kmm-ci-build.template.yaml [hunk 2]
```diff
@@ -23,9 +23,8 @@ spec:
               - name: build-secret
             dockerfileConfigMap:
               name: kmm-kmod-dockerfile
-            
             # Optional. If kanikoParams.tag is empty, the default value will be: 'latest'
             kanikoParams:
               tag: "debug"
   selector:
-    kubernetes.io/hostname: minikube
+    kubernetes.io/hostname: kmm-ci
```
Comment (qbarrand) [inline ci/module-kmm-ci-build.template.yaml:30]: Do we have such a node on the test cluster?

Comment (enriquebelarte) [reply ci/module-kmm-ci-build.template.yaml:30]: I pretended to label the node before applying:

https://github.com/rh-ecosystem-edge/kernel-module-management/pull/267/files#diff-d954caf0668f1d5539f11597234490aa6aadf4e53c4e175ba384b11244dbd21eR12

Comment (qbarrand) [reply ci/module-kmm-ci-build.template.yaml:30]: I think we should use another label key then. `kubernetes.io/hostname` is managed by Kubernetes and might be overwritten.

**Review by qbarrand — APPROVED** [review]

**Review by qbarrand — APPROVED** [review]

**Pull request merged**